Makefile

```diff
@@ -24,456 +24,57 @@
-# examples
-
-# example of static JS compilation
-HELLO_SRCS=examples/hello.js
-HELLO_OPTS=-fno-string-normalize -fno-map -fno-promise -fno-typedarray \
-           -fno-typedarray -fno-regexp -fno-json -fno-eval -fno-proxy \
-           -fno-date -fno-module-loader
-
-hello.c: $(QJSC) $(HELLO_SRCS)
-	$(QJSC) -e $(HELLO_OPTS) -o $@ $(HELLO_SRCS)
-
-ifdef CONFIG_M32
-examples/hello: $(OBJDIR)/hello.m32s.o $(patsubst %.o, %.m32s.o, $(QJS_LIB_OBJS))
-	$(CC) -m32 $(LDFLAGS) -o $@ $^ $(LIBS)
-else
-examples/hello: $(OBJDIR)/hello.o $(QJS_LIB_OBJS)
-	$(CC) $(LDFLAGS) -o $@ $^ $(LIBS)
-endif
-
-# example of static JS compilation with modules
-HELLO_MODULE_SRCS=examples/hello_module.js
-HELLO_MODULE_OPTS=-fno-string-normalize -fno-map -fno-typedarray \
-           -fno-typedarray -fno-regexp -fno-json -fno-eval -fno-proxy \
-           -fno-date -m
-examples/hello_module: $(QJSC) libquickjs$(LTOEXT).a $(HELLO_MODULE_SRCS)
-	$(QJSC) $(HELLO_MODULE_OPTS) -o $@ $(HELLO_MODULE_SRCS)
-
-# use of an external C module (static compilation)
-
-test_fib.c: $(QJSC) examples/test_fib.js
-	$(QJSC) -e -M examples/fib.so,fib -m -o $@ examples/test_fib.js
-
-examples/test_fib: $(OBJDIR)/test_fib.o $(OBJDIR)/examples/fib.o libquickjs$(LTOEXT).a
-	$(CC) $(LDFLAGS) -o $@ $^ $(LIBS)
-
-examples/fib.so: $(OBJDIR)/examples/fib.pic.o
-	$(CC) $(LDFLAGS) -shared -o $@ $^
-
-examples/point.so: $(OBJDIR)/examples/point.pic.o
-	$(CC) $(LDFLAGS) -shared -o $@ $^
-
-###############################################################################
-# documentation
-
-DOCS=doc/quickjs.pdf doc/quickjs.html
-
-build_doc: $(DOCS)
-
-clean_doc: 
-	rm -f $(DOCS)
-
-doc/%.pdf: doc/%.texi
-	texi2pdf --clean -o $@ -q $<
-
-doc/%.html.pre: doc/%.texi
-	makeinfo --html --no-headers --no-split --number-sections -o $@ $<
-
-doc/%.html: doc/%.html.pre
-	sed -e 's|</style>|</style>\n<meta name="viewport" content="width=device-width, initial-scale=1.0">|' < $< > $@
-
-###############################################################################
-# tests
-
-ifndef CONFIG_MINGW
-ifndef CONFIG_DARWIN
-test: tests/bjson.so examples/point.so
-endif
-endif
-ifdef CONFIG_M32
-test: qjs32
-endif
-
-test: qjs
-	./qjs tests/test_bigint.js
-	./qjs tests/test_closure.js
-	./qjs tests/test_language.js
-	./qjs tests/test_builtin.js
-	./qjs tests/test_loop.js
-	./qjs tests/test_std.js
-ifndef CONFIG_MINGW
-	./qjs tests/test_worker.js
-endif
-ifndef CONFIG_MINGW
-ifndef CONFIG_DARWIN
-	./qjs tests/test_bjson.js
-	./qjs examples/test_point.js
-endif
-endif
-ifdef CONFIG_M32
-	./qjs32 tests/test_closure.js
-	./qjs32 tests/test_language.js
-	./qjs32 tests/test_builtin.js
-	./qjs32 tests/test_loop.js
-	./qjs32 tests/test_std.js
-	./qjs32 tests/test_worker.js
-	./qjs32 tests/test_bigint.js
-endif
-
-stats: qjs qjs32
-	./qjs -qd
-	./qjs32 -qd
-
-microbench: qjs
-	./qjs tests/microbench.js
-
-microbench-32: qjs32
-	./qjs32 tests/microbench.js
-
-# ES5 tests (obsolete)
```

Comment:
I can restore this if we want, but I guess we don't?

Reply:
I agree.